test/integration/UniqueParticle.spec.ts

```diff
@@ -0,0 +1,194 @@
+            logger.error(ERROR_MESSAGE)
+            throw new Error(ERROR_MESSAGE)
+        }
+
+        await identity1.account.openNodeConnection()
+
+
+        // Create token
+
+        const symbol = 'UNIQ'
+        const name = 'UNIQ test'
+        const description = 'my token description'
+        const granularity = new BN(1)
+        const amount = 1000
+
+        await new RadixTransactionBuilder().createTokenMultiIssuance(
+            identity1.account,
+            name,
+            symbol,
+            description,
+            granularity,
+            amount,
+        )
+        .signAndSubmit(identity1)
+        .toPromise()
+    })
+
+    after(async () => {
+        await identity1.account.closeNodeConnection()
+    })
+
+    it('(1) should create an atom with a unique id', function (done) {
+        this.timeout(50000)
+
+        RadixTransactionBuilder.createMintAtom(
+            identity1.account,
+            testTokenRef,
+            1)
+        .addUniqueParticle(identity1.account, 'unique1')
+        .signAndSubmit(identity1)
+        .subscribe({
+            complete: () => {
+                done()
+            },
+            error: e => done(new Error(JSON.stringify(e))),
+        })
+    })
+
+    it('TODO: it should fail when submitting a unique particle for an unowned account', function (done) {
+        this.timeout(50000)
+
+        RadixTransactionBuilder.createMintAtom(
+            identity1.account,
+            testTokenRef,
+            1)
+        .addUniqueParticle(identity2.account, 'unique1')
+        .signAndSubmit(identity1)
+        .subscribe({
+            complete: () => {
+                done()
+            },
+            error: e => done(new Error(JSON.stringify(e))),
+        })
+    })
+
+    it('(2) should fail submitting an atom with a conflicing unique id', function (done) {
+        this.timeout(50000)
+
+        RadixTransactionBuilder.createMintAtom(
+            identity1.account,
+            testTokenRef,
+            1)
+        .addUniqueParticle(identity1.account, 'unique1')
+        .signAndSubmit(identity1)
+        .subscribe({
+            complete: () => {
+                done('Should have failed')
+            },
+            error: e => {
+                expect(e).to.contain('unique require compromised')
+                done()
+            },
+        })
+    })
+
+    it('(3) should succeed submitting an atom with multiple unique ids', function (done) {
+        this.timeout(50000)
+
+        RadixTransactionBuilder.createMintAtom(
+            identity1.account,
+            testTokenRef,
+            1)
+        .addUniqueParticle(identity1.account, 'unique2')
+        .addUniqueParticle(identity1.account, 'unique3')
+        .signAndSubmit(identity1)
+        .subscribe({
+            complete: () => {
+                // Check balance
+                done()
+            },
+            error: e => done(new Error(JSON.stringify(e))),
+        })
+    })
```

Comment:
What would happen if we ran this test directly after the above one?

```javascript
    it('(3b) should succeed/fail submitting an atom with partially conflicting unique id', function (done) {
        this.timeout(50000)

         RadixTransactionBuilder.createMintAtom(
            identity1.account,
            testTokenRef,
            1)
        .addUniqueParticle(identity1.account, 'unique2')
        .signAndSubmit(identity1)
        .subscribe({
            complete: () => {
                // Check balance
                done()
            },
            error: e => done(new Error(JSON.stringify(e))),
        })
    })
```

I guess my real question is whether it's the combination of `unique`'s that is unique or each individual `unique`.

Reply:
It's each individual unique, so your test should fail. I'll add it as a real test